src/components/VmDetails/cards/DisksCard/DiskImageEditor.js

```diff
@@ -155,47 +148,59 @@ class DiskImageEditor extends Component {
     this.setState({ showModal: false })
   }
 
-  // NOTE: Add and edit both can use the same composeDisk() since the add and edit
-  //       sagas will use what they need from the composed disk.  The sagas ultimately
-  //       control the REST calls and data.
   composeDisk () {
-    const { vm, disk } = this.props
-    const vmDiskInSameStorageDomain =
-      vm.get('disks') &&
-      vm.get('disks').find(disk => disk.get('storageDomainId') === this.state.values.storageDomain)
-
-    const iface =
-      (disk && disk.get('iface')) ||
-      (vmDiskInSameStorageDomain && vmDiskInSameStorageDomain.get('iface')) ||
-      'virtio_scsi'
-
-    const provisionedSize = disk
-      ? disk.get('provisionedSize') + this.state.values.size * 1024 ** 3
-      : this.state.values.size * 1024 ** 3
-
-    const bootable = this.state.values.bootable
-
-    const newDisk = {
-      ...DISK_DEFAULTS,
-      attachmentId: disk && disk.get('attachmentId'),
-      storageDomainId: this.state.values.storageDomain,
-
-      iface,
-      id: this.state.id,
-      name: this.state.values.alias,
-      provisionedSize,
-      bootable,
-
-      // the __diskType__ field maps to format + sparse REST Disk attributes
-      format: 'raw', // Match webadmin behavior, disks are created as 'raw'
-      sparse: this.state.values.diskType === 'thin',
-    }
+    const { vm, disk, storageDomains } = this.props
```

Comment:
>   1. there is no code re-use between `new` and 'edit` - how about extracting those cases to separate functions?

Ok.  `composeDiskEdit()` and `composeNewDisk()`

> 
>   2. can we unit test this logic?

I don't want to do that in this PR since they're tied to the component's data etc.

Reply:
>  can we unit test this logic?
>> I don't want to do that in this PR since they're tied to the component's data etc.

OK.  Some refactoring would be needed to pass all the data via parameters. Some follow-up PR would be better here.